hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreExtension.java

```diff
@@ -80,4 +82,27 @@ public HiveConf hiveConf() {
   public TestHiveMetastore metastore() {
     return metastore;
   }
+
+  public static Builder builder() {
```

Comment:
Just a constructor overloading was enough? 
It can also helps in avoiding the code changes in existing constructor callers?

Reply:
I'm not sure I fully understand the question, but we typically try to avoid adding (multiple) parameters to the builder method as otherwise this is just mimicking the constructor

Reply:
I see Ajantha's point, but since it is in tests I don't think that it is really a problem of hiding the constructor, also this makes it more futureproof.

Reply:
I meant have two constructor 
HiveMetastoreExtension(DB_NAME, Collections.emptyMap()) -- existing
HiveMetastoreExtension(Collections.emptyMap()) -- new

If DB_NAME is not passed, don't initialize the db. 

If we have multiple params, builder makes sense. But for just one param it might be an overkill. 

But I don't have strong oppose to the current changes. We can merge it.

Reply:
I think that using a Builder is more future-proof instead of having multiple constructors here